examples/timestamping/backend/tests/config.rs

```diff
@@ -0,0 +1,137 @@
+                    time_service_name: TIME_SERVICE_NAME.to_owned(),
+                }),
+        );
+
+    let testkit = if second_time_service {
+        let time_service = TimeServiceFactory::with_provider(mock_provider.clone());
+        let time_service_artifact = time_service.artifact_id();
+        testkit
+            .with_rust_service(time_service)
+            .with_artifact(time_service_artifact.clone())
+            .with_instance(
+                time_service_artifact
+                    .into_default_instance(SECOND_TIME_SERVICE_ID, SECOND_TIME_SERVICE_NAME),
+            )
+            .build()
+    } else {
+        testkit.build()
+    };
+
+    (testkit, mock_provider)
+}
+
+/// Creates block with `ConfigPropose` tx and returns current service configuration.
+fn propose_configuration(testkit: &mut TestKit, config: Config) -> Config {
+    let tx = ConfigPropose::immediate(0).service_config(SERVICE_ID, config);
+
+    let initiator_id = testkit.network().us().validator_id().unwrap();
+    let (pub_key, sec_key) = &testkit.validator(initiator_id).service_keypair();
+    testkit.create_block_with_transaction(tx.sign_for_supervisor(*pub_key, sec_key));
+
+    testkit
+        .snapshot()
+        .for_service(SERVICE_NAME)
+        .unwrap()
+        .get_entry("config")
+        .get()
+        .unwrap()
+}
+
+#[test]
+fn test_propose_configuration() {
+    let (mut testkit, _) = init_testkit(true);
+    let config = Config {
+        time_service_name: SECOND_TIME_SERVICE_NAME.to_string(),
+    };
+
+    // Propose valid configuration.
+    let new_config = propose_configuration(&mut testkit, config.clone());
+
+    assert_eq!(new_config.time_service_name, config.time_service_name);
+}
+
+#[test]
+fn test_propose_invalid_configuration() {
+    // Empty service name.
+    let (mut testkit, _) = init_testkit(false);
+    let config = Config {
+        time_service_name: "".to_string(),
+    };
+    // Propose configuration with invalid time service name.
+    let new_config = propose_configuration(&mut testkit, config);
+    // Check that configuration has not changed.
+    assert_eq!(new_config.time_service_name, TIME_SERVICE_NAME);
+
+    // Service name with illegal character.
+    let config = Config {
+        time_service_name: "illegal.illegal".to_string(),
+    };
+    // Propose configuration with not existing service name.
+    let new_config = propose_configuration(&mut testkit, config.clone());
+    // Check that configuration has not changed.
+    assert_eq!(new_config.time_service_name, TIME_SERVICE_NAME);
+
+    // Name of not existing service.
+    let config = Config {
+        time_service_name: "not_service".to_string(),
+    };
+    // Propose configuration with not existing service name.
+    let new_config = propose_configuration(&mut testkit, config.clone());
+    // Check that configuration has not changed.
+    assert_eq!(new_config.time_service_name, TIME_SERVICE_NAME);
```

Comment:
With the tightened checks I've proposed above, it would be possible to test that the service refuses a configuration with a service name that corresponds to a bogus artifact, for example, `SERVICE_NAME`.